### PR TITLE
feat: Add leader election metrics

### PR DESCRIPTION
**Pull request by chrkl**
## Description

Changes proposed in this pull request (what was done and why):

- Add leader election metrics

Changes refer to particular issues, PRs or documents:

- #25 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The PR has a milestone set.
- [x] The PR has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [ ] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

**Review by NHingerl — DISMISSED** [review]

File: receiver/singletonreceivercreator/factory.go [hunk 1]
```diff
@@ -35,8 +36,26 @@ func createDefaultConfig() component.Config {
 	}
 }
 
-func createMetricsReceiver(_ context.Context, params receiver.Settings, cfg component.Config, consumer consumer.Metrics) (receiver.Metrics, error) {
-	r := newSingletonReceiverCreator(params, cfg.(*Config))
-	r.nextMetricsConsumer = consumer
-	return r, nil
+func createMetricsReceiver(
```
Comment (rakesh-garimella) [inline receiver/singletonreceivercreator/factory.go:39]: the signature format is different from rest of the repo

File: receiver/singletonreceivercreator/leader_elector.go [hunk 2]
```diff
@@ -36,7 +41,13 @@ func newResourceLock(client kubernetes.Interface, leaderElectionNamespace, lockN
 }
 
 // newLeaderElector return  a leader elector object using client-go
-func newLeaderElector(client kubernetes.Interface, onStartedLeading func(context.Context), onStoppedLeading func(), cfg leaderElectionConfig) (*leaderelection.LeaderElector, error) {
+func newLeaderElector(
```
Comment (rakesh-garimella) [inline receiver/singletonreceivercreator/leader_elector.go:44]: signature format different from rest of the project

File: receiver/singletonreceivercreator/receiver.go [hunk 3]
```diff
@@ -8,23 +8,33 @@ import (
...
 	nextMetricsConsumer consumer.Metrics
+	telemetryBuilder    *metadata.TelemetryBuilder
 
 	host              component.Host
 	subReceiverRunner *receiverRunner
 	cancel            context.CancelFunc
 }
 
-func newSingletonReceiverCreator(params receiver.Settings, cfg *Config) *singletonReceiverCreator {
+func newSingletonReceiverCreator(
```
Comment (rakesh-garimella) [inline receiver/singletonreceivercreator/receiver.go:27]: function signature format

**Review by rakesh-garimella — DISMISSED** [review]

File: receiver/singletonreceivercreator/factory.go (hunk 1 above)
Comment (skhalash) [reply receiver/singletonreceivercreator/factory.go:39]: As discussed, this format is pretty common (we are even using it in telemetry-manager) if the argument list gets too long, which is often the case for ctor functions (`newXxx`)

File: receiver/singletonreceivercreator/documentation.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+# singleton_receiver_creator
+
+## Internal Telemetry
+
+The following telemetry is emitted by this component.
+
+### receiver_singleton_leader_status
+
+Gauge of if the reporting system is leader of the relevant lease, 0 indicates backup, 1 indicates leader.
```
Comment (mmitoraj) [inline receiver/singletonreceivercreator/documentation.md:11]:
```suggestion
A gauge of if the reporting system is the leader of the relevant lease, 0 indicates backup, and 1 indicates leader.
```

File: receiver/singletonreceivercreator/documentation.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Total number of lease losses.
+
+| Unit | Metric Type | Value Type | Monotonic |
+| ---- | ----------- | ---------- | --------- |
+| 1 | Sum | Int | true |
+
+### receiver_singleton_lease_slowpath_total
+
+Total number of slow path exercised in renewing leader leases.
```
Comment (mmitoraj) [inline receiver/singletonreceivercreator/documentation.md:35]:
```suggestion
The total number of slow paths exercised in renewing leader leases.
```

File: receiver/singletonreceivercreator/documentation.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+Total number of successful lease acquisitions.
+
+| Unit | Metric Type | Value Type | Monotonic |
+| ---- | ----------- | ---------- | --------- |
+| 1 | Sum | Int | true |
+
+### receiver_singleton_lease_lost_total
+
+Total number of lease losses.
```
Comment (mmitoraj) [inline receiver/singletonreceivercreator/documentation.md:27]:
```suggestion
The total number of lease losses.
```

File: receiver/singletonreceivercreator/documentation.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+Gauge of if the reporting system is leader of the relevant lease, 0 indicates backup, 1 indicates leader.
+
+| Unit | Metric Type | Value Type |
+| ---- | ----------- | ---------- |
+| 1 | Gauge | Int |
+
+### receiver_singleton_lease_acquired_total
+
+Total number of successful lease acquisitions.
```
Comment (mmitoraj) [inline receiver/singletonreceivercreator/documentation.md:19]:
```suggestion
The total number of successful lease acquisitions.
```

**Review by mmitoraj — APPROVED** [review]

**Review by rakesh-garimella — APPROVED** [review]

**Pull request merged**